shim-server/src/main/java/org/openmhealth/shimmer/common/controller/LegacyAuthorizationController.java

```diff
@@ -211,22 +237,35 @@ public AuthorizationRequestParameters initiateAuthorization(
         if (authParams.getClientRedirectUrl() != null) {
             try {
                 servletResponse.sendRedirect(authParams.getClientRedirectUrl());
-            }
-            catch (IOException e) {
+            } catch (IOException e) {
                 e.printStackTrace();
-                throw new ShimException("Error occurred redirecting to :" + authParams.getRedirectUri());
+                throw new ShimException(
+                    "Error occurred redirecting to :" + authParams.getRedirectUri());
             }
             return null;
         }
 
         return ok(response);
     }
 
+    @PostMapping(value = "/authorize")
```

Comment:
This is the end point and DTO that i have added. The rest are formatting edits.

Reply:
If this is completely independent from private methods, would suggest adding a new Controller class and inject `accessParametersRepo` there for your usage.

Reply:
All the '/authorize' URLs are handled by this controller. That's why have added this end point here. Should I change?